### DISCUSSION TITLE
Support Database Read Replicas

### DISCUSSION THREAD

**Discussion by jamescjensen** [medusajs/medusa#7061] [Feature Requests]
Please correct me if I'm wrong, but Medusa currently doesn't seem to support read replicas. `packages/medusa/src/loaders/database.ts` has a hard config for how the database should be set up, only allowing a single URL.

Typeorm [does support read replicas](https://orkhan.gitbook.io/typeorm/docs/multiple-data-sources#replication). I think the solution would be to let users pass more params to the `new DataSource` call instead of the currently limited set of DataSourceOptions used by `loaders/database.ts`.

**Comment by olivermrbl** [comment]
That is correct, we currently don't support a read replica setup. As you note, the solution is seemingly simple, so we are open to adding this in. Only thing stopping us from doing it right now is that we don't have time to run proper testing. You are more than welcome to send a PR and run these tests yourself, if you need this urgently.

We will support it in v2.0, releasing soon.
